### DISCUSSION TITLE
有关exit指令的问题

### DISCUSSION THREAD

**Discussion by backrunner** [koishijs/koishi#265] [Q&A]
之前我的bot是用v1写的，这几天开始用v3 (3.10.2) 写一个新的bot，目前我对v3还不是很熟悉。

因为bot的功能比较简单，我没有接数据库存用户信息的打算，只想单纯运行这个bot。
在使用过程中我发现v3默认添加了一个exit指令，它默认出现在了help里面，我试了一下可以直接执行，koishi的进程直接退出了...

翻文档的时候我找到这个指令来自于koishi-plugin-common，且这个是默认安装的，但我在我的node_modules里找不到这个包，在koishi.config.js里按照下面这样设置：
```js
plugins: {
  'common': {
    exit: false,
  },
},
```
启动bot之后会直接报Error: cannot resolve plugin common（因为这个包根本不存在）。

翻看源码之后我发现这个指令是在`koishi/packages/koishi/src/worker.ts`注册的，文档没有提及这一点，代码看上去也没办法不注册这个指令（或者在生产环境下不注册）。虽然它的权限设置为了4，但不知道是不是因为我没有接数据库，导致权限失效了（但是文档提到用户默认的权限应该是0，我没有设置任何用户权限方面的东西，却直接执行了exit，很迷惑）。

我不确定是我的配置问题导致权限失效了，还是代码存在问题，目前我只能临时把注册exit指令的那一段代码删掉来避免我的用户触发它...

即使这个指令在权限方面是没问题的，我个人希望它默认可以设置为隐藏。

**Comment by shigma** [comment]
你说的问题我之前没有考虑到，感谢反馈。未来一个版本将作出以下修改：

1. 当未配置数据库时，exit 指令将默认隐藏，可通过配置项进行配置
2. 完善 common 插件和命令行工具的文档

**Comment by idlist** [comment]
如果你是想删掉这个指令的话，有一个简单的方法（也是我在用的）

```js
ctx.dispose('exit')
```

如果使用cli的话可以考虑写一个本地插件（比如 [`plugin-tweaks.js`](https://github.com/idlist/2bot-v3/blob/master/plugin-tweaks.js) ） 然后把上面这行放进来以修改行为
实际上好像只有用cli启动的时候才会添加这个指令，用index启动的时候是不会有的（印象中，不一定对）
